src/app/search/search-filters/wishlist.tsx

```diff
@@ -9,7 +9,15 @@ const checkIfIsWishlist = (
   wishListFunction: (item: DimItem) => InventoryWishListRoll | undefined
 ) => {
   const roll = wishListFunction(item);
-  return roll && !roll.isUndesirable;
+  return roll && !roll.isUndesirable && !roll.isUnknown;
+};
+
+const checkIfIsWishlistUnknown = (
+  item: DimItem,
+  wishListFunction: (item: DimItem) => InventoryWishListRoll | undefined
+) => {
+  const roll = wishListFunction(item);
+  return roll?.isUnknown;
```

Comment:
Can we not use the absence of a wishlist roll to determine that there is no wish list roll for the item? e.g.:

```suggestion
  return roll === undefined;
```

Reply:
The problem is that right now the selector returns null in two distinct cases: 
* It found a gun with different perks than what you asked for
* It didn't find anything.
This is what I was talking about where I said using null loses information. Higher up in the filtering and UI, we don't know which case null means. For the UI we mostly just care about Good or Bad, but in the filtering we want to disambiguate the difference so that `is:wishlistunknown` returns different results than `not:wishlist` (as the original issue discussion brought up.)
"Neutral" in my other implementation meant "found a gun, didn't like the perks" whereas "Unknown" meant didn't find anything at all...
This gives the four states that we (now) care about:
* Found in wishlist
* * Good Roll
* * Bad Roll
* Not Found
* * Meh Roll (Neutral)
* * Never heard of it

Anyway, since we use the same object in filtering and in the UI, we need a way to support these two use cases and the selector methods don't know which use case is being satisfied, so I made it handle either with the new flag, (or in the other implementation always having a return object and being explicit about the results.)

Reply:
Thanks! I wish this stuff was documented so it'd be easier to understand all that.

I suppose what I was thinking when I originally thought about this task was that we wouldn't use the same wish list roll object as we use for display - we'd just check if there are any wish list rolls for the item hash at all, and use that in the filter.